extension/packagemanagerextension/nbextension/methods.js

```diff
@@ -0,0 +1,148 @@
+
+define([
+    'jquery',
+    './common',
+    './api',
+], function ($, common, api) {
+    "use strict";
+
+    /*
+    This function returns the latest version of a package.
+    */
+
+    function check_new_version(updates, packagename, pkgversion) {
+        for (let element of updates) {
+            if (element.name === packagename) {
+                return element.version;
+            }
+        }
+        return pkgversion;
+    }
+
+    /*
+    This function lists down all the updates and requires a user confirmation.
+    */
+
+    async function update_packages(project) {
+        $('#updatebtn').unbind();
+        $('#updatebtn').click(async function () {
+            let selectedPackages = common.get_selected_packages();
+            let html = "<p> The following packages are about to be updated: </p> </br>";
+            $('#updatebtn').toggleClass('fa-wrench fa-spinner').addClass('fa-spin');
+            let resp = await api.check_update(project);
+            $('#updatebtn').toggleClass('fa-wrench fa-spinner').removeClass('fa-spin');
+            let updates = resp.updates;
+            let packages = [];
+            let ct = 0;
+
+            if (selectedPackages.length > 0) {
+                html += "<ul>";
```

Comment:
this was not converted to jquery... And this was the part where it is needed the most.